packages/core/src/elements/FormFieldElement.ts

```diff
@@ -0,0 +1,178 @@
+    }
+
+    if (ObservedAriaDescription.includes(name)) {
+      this.requestUpdate(AriaDescriptionKey, Symbol());
+    }
+
+    if (ObservedAriaRequired.includes(name)) {
+      this.requestUpdate(AriaRequiredKey, Symbol());
+    }
+  }
+
+  /**
+   * Compute property values that depend on other properties
+   * and are used in the rest of the update process.
+   * @param changedProperties Properties that has changed
+   * @returns {void}
+   */
+  public willUpdate (changedProperties: PropertyValues): void {
+    super.willUpdate(changedProperties);
+
+    if (changedProperties.has(AriaLabelKey)) {
+      this.inputAriaLabel = inputLabel(this);
+    }
+
+    if (changedProperties.has(AriaRequiredKey)) {
+      this.inputAriaRequired = inputRequired(this);
+    }
+
+    // Description is traditionally changed with error, so recalculate on error changed as well
+    if (changedProperties.has(AriaDescriptionKey)
+      || (changedProperties.get('error') !== undefined && changedProperties.has('error'))
+    ) {
+      this.inputAriaDescription = inputDescription(this);
+    }
+  }
+
+  /**
+   * Called after the component is first rendered
+   * @param changedProperties Properties which have changed
+   * @returns {void}
+   */
+  protected firstUpdated (changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+    this.addEventListener('focus', this.onFocus);
+  }
+
+  /**
+   * Handles the focus event
+   * @param event Focus event
+   * @returns {void}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected onFocus (event: FocusEvent): void {
+    // Always recalculate inherited arias on focus and these may change from last time
+    this.setInheritedAria();
+  }
+
+  /**
+   * Set inherited aria properties
+   * @returns {void}
+   */
+  protected setInheritedAria (): void {
+    this.inputAriaLabel = inputLabel(this);
+    this.inputAriaDescription = inputDescription(this);
+    // no need for required or error here, as it is sufficient to recalculate these on attributeChanged
+  }
+
+  /**
+   * Decorate `<input>` element with aria properties:
+   * aria-hidden="true" - always true. Required for onLoad screen read. this is always true for Chrome. Other browsers may need to tweak
+   * aria-label - calculated from `aria-label`, `aria-labelledby` and `label[for="<element.id>"]`
+   * aria-description - calculated from `aria-description` or `aria-describedby`
+   * required - calculated on based on `aria-required` and `required` state
+   * aria-invalid="true|false" - calculated on based on `error` state
+   * @returns noChange, directive result
+   */
+  protected ariaDecorate (): AttributeMapDirectiveResult {
+    return attributeMap({
+      'aria-hidden': 'true',
+      'aria-label': this.inputAriaLabel,
+      'aria-description': this.inputAriaDescription,
+      'aria-invalid': this.error ? 'true' : 'false',
+      required: this.inputAriaRequired
```

Comment:
`required` shouldn't be used to avoid confusing users as we do not support native required functionality in our custom text-field, we should use only `aria-required`.

Reply:
Done